errors/validation_errors.go

```diff
@@ -1,10 +1,44 @@
 package errors
 
-type ValidationErrors map[string][]string
+import "fmt"
 
-func (s ValidationErrors) Add(field string, message string) {
-	if _, ok := s[field]; !ok {
-		s[field] = make([]string, 0)
+var _ error = ValidationError{}
+
+type ValidationError struct {
+	Context string
+	Message string
+}
+
+func (ve ValidationError) Error() string {
+	return fmt.Sprintf("%s - %s", ve.Context, ve.Message)
+}
+
+var _ error = ValidationErrors{}
+
+type ValidationErrors []ValidationError
+
+func (ve ValidationErrors) Error() string {
+	result := ""
+	if len(ve) > 0 {
+		result += "validation errors:\n"
+	}
+	for _, err := range ve {
+		result += fmt.Sprintf("\t%s %s\n", err.Context, err.Error())
+	}
+	return result
+}
+
+func (ve ValidationErrors) ToJson() map[string][]string {
```

Comment:
I expected `ToJson()` to emit a json string. Perhaps, `ToMap()` would be a better name.
```suggestion
func (ve ValidationErrors) ToMap() map[string][]string {
```